### PR TITLE
Change dotnet TargetFramework from v7.0 to v8.0

### PR DESCRIPTION
**Pull request by pngan**
# Description

Updated all the csproj files to use dotnet v8.

## Issue reference

We strive to have all PRs being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ x] The quickstart code compiles correctly
* [ x] You've tested new builds of the quickstart if you changed quickstart code
* [x ] You've updated the quickstart's README if necessary
* [ ] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

**Comment by pngan** [comment]
Hi Could someone double check this pull request. Currently DCO has failed, but I updated the commit message to have a `Signed-off-by:` section, and put myself as the sign off person. Could someone fix this for me, if I have done something wrong. Thanks.

**Review by paulyuk — APPROVED** [review]
LGTM.  Thank you for this!!

**Review by paulyuk — CHANGES_REQUESTED** [review]
Now that I think more about it, may I ask for a few quick changes that make this work from end to end?

1) Add .NET 8 as a second version to the validate csharp workflow here:
```json
            7.0.x
            8.0.x
```
https://github.com/dapr/quickstarts/blob/release-1.13/.github/workflows/validate_csharp_quicstarts.yaml#L82
2) omit all files in http/batch/Debug/net6.0
3) make the change for Workflow quickstarts too
4) change this PR to baseline against the `release-1.13` branch instead of `master` so we keep this aligned with this latest release.

I'm happy to help with any of this, but most of these changes have to start from your side or the files will not be included in the PR.  thank you!

File: bindings/csharp/http/batch/Debug/net6.0/batch.runtimeconfig.json
Comment (paulyuk) [inline bindings/csharp/http/batch/Debug/net6.0/batch.runtimeconfig.json:1]: Please exclude this from the PR, and double check your gitignore on all /Debug or /bin folders

**Comment by paulyuk** [comment]
Fixes #980

**Comment by pngan** [comment]
This pull request is superseded by pr 980.

**Pull request closed without merging**